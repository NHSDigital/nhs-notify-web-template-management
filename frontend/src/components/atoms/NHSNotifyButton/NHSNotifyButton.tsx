'use client';

import { Button } from 'nhsuk-react-components';
import { ButtonType } from './button.types';

export function NHSNotifyButton({
  children,
  href,
  ...rest
}: ButtonType & React.ComponentProps<typeof Button>) {
  return (
    <Button preventDoubleClick {...rest} href={href} data-testid='link-button'>
      {children}
    </Button>
  );
}
