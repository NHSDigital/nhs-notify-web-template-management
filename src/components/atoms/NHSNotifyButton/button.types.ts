export type ButtonType = {
  children: React.ReactNode;
  href?: string;
  className?: string;
  onClick?: () => void;
};
