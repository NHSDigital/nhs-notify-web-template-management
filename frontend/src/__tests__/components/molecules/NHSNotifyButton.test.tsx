import { fireEvent, render, screen } from '@testing-library/react';
import { NHSNotifyButton } from '@atoms/NHSNotifyButton/NHSNotifyButton';

describe('NHS Notify button', () => {
  it('renders component correctly as a button', () => {
    render(
      <NHSNotifyButton data-testid='button'> Button text</NHSNotifyButton>
    );

    expect(screen.getByTestId('button')).toBeInTheDocument();
    expect(screen.getByTestId('button')).toHaveTextContent('Button text');
  });

  it('renders component correctly as a link button', () => {
    render(
      <NHSNotifyButton data-testid='link-button' href='#'>
        {' '}
        Button text
      </NHSNotifyButton>
    );

    expect(screen.getByTestId('link-button')).toBeInTheDocument();
    expect(screen.getByTestId('link-button')).toHaveTextContent('Button text');
    expect(screen.getByTestId('link-button')).toHaveAttribute('href', '#');
  });

  it('debounces multiple clicks', async () => {
    const onClick = jest.fn();

    render(
      <NHSNotifyButton data-testid='button' onClick={onClick}>
        {' '}
        Button text
      </NHSNotifyButton>
    );

    const button = screen.getByTestId('button');

    fireEvent.click(button);
    fireEvent.click(button);
    fireEvent.click(button);
    fireEvent.click(button);
    fireEvent.click(button);
    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('default onClick does nothing', async () => {
    const container = render(
      <NHSNotifyButton data-testid='button'>Button text</NHSNotifyButton>
    );

    const button = screen.getByTestId('button');

    fireEvent.click(button);

    expect(container.asFragment()).toMatchSnapshot();
  });
});
