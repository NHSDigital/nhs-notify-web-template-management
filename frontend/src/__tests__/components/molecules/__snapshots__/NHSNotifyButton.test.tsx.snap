// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NHS Notify button default onClick does nothing 1`] = `
<DocumentFragment>
  <button
    aria-disabled="false"
    class="nhsuk-button"
    data-testid="button"
    type="submit"
  >
    Button text
  </button>
</DocumentFragment>
`;
